### ISSUE TITLE
Resynthesizer plugin fails to load due to undefined symbol error

### ISSUE THREAD

**Issue by neildarlow** [flathub/org.gimp.GIMP#368]
After installing GIMP-3.0 RC1 and its associated Resynthesizer plugin from flathub-beta, the plugin falils to load at startup.

Terminal output as follows:
`
neil@vivobook:~$ flatpak run org.gimp.GIMP
gimp_font_factory_load_names: 99 unsupported fonts were ignored. Set the GIMP_DEBUG_FONTS environment variable for a listing.
set device 'Wayland Pointer' to mode: disabled
/app/extensions/plug-ins/Resynthesizer/Resynthesizer: symbol lookup error: /app/extensions/plug-ins/Resynthesizer/Resynthesizer: undefined symbol: gimp_procedure_add_argument
gimp-3.0: LibGimpBase-WARNING: gimp-3.0: gimp_wire_read(): unexpected EOF
`
Am I right in thinking the plugin is looking for gimp_procedure_add_argument (which was added to the 3.0 API)? In which case, is the build missing a dependency?

**Comment by hfiguiere** [comment]
well actually this is just the wrong repo. See https://github.com/flathub/org.gimp.GIMP.Plugin.Resynthesizer

But the problem is that the plugin API broke and it needs to be updated by upstream before it can be rebuilt.

**Comment by brunvonlope** [comment]
> well actually this is just the wrong repo

Nope. If the code of the plug-in wasn't updated, there is nothing that you can do downstream in the manifest (patching doesn't counts).

The manifest clearly clones the repo that I linked so there is what everything begins to Resyntethizer. Of course, it's possible to report downstream but I'm not sure that @bootchk will be notified.

**Comment by hfiguiere** [comment]
> > well actually this is just the wrong repo
> 
> Nope. If the **code** of the plug-in wasn't updated, there is nothing that you can do downstream in the manifest (patching doesn't counts).

I will repeat again. **The issue was filed in the wrong repo.** The extension has been packaged for the gimp 3 branch and worked with the previous release. Hence the issue should have been filed there. Now if I forget about it that will be the reason.

GMIC is on the same boat, but for longer.

> The manifest clearly clones the repo 

Tell me about the manifest. I can explain what I did: I cloned the branch that has the GIMP 3 port. It's also on flathub-beta.

**Comment by brunvonlope** [comment]
@hfiguiere What I was saying is that: the fact of the plug-in isn't working much likely it is NOT your "fault" as a packager.

Indeed I didn't paid attention that there is another branch, more updated, but this doesn't change the fact that the code of that branch is from August and too much things changed in GIMP code in the last days before RC1. Even if it was updated to a further commit in the manifesf, probably Resyntehtizer wouldn't work. That's my theory, guard down.

**Comment by neildarlow** [comment]
Alright, my bad for reporting in the wrong place but, in my defence, ...

The flathub.org web interface presents the application and its associated plugins with a drop-down to select where you install from (distribution repo, flathub or flathub-beta). The application is listed with website and issue reporting links but plugins are just listed by name with an associated install (or uninstall) button.

Furthermore, it appears that plugins are listed globally for stable and beta versions on the same page whether they are appropriate to beta versions or not. This is a failing of the flathub.org interface.

Consider that the GIMP developers see 3.0 RC1 as suitable for a wider audience to test and they publish install links to the flathub-beta repo. Other users will install by visiting the flathub.org page and selecting the beta install. And where will they go to report issues found? They will follow the issue reporting link on that page.

Expect more of this to occur. Now, if incompatible plugins weren't shown when a beta install is selected then we'd be in a totally different place.
